src/commands/math/commands.js

```diff
@@ -1242,8 +1242,12 @@ LatexCmds.MathQuillVarField = P(MathCommand, function(_, super_) {
     +   '<span class="mq-root-block">&0</span>'
     + '</span>'
   ;
-  _.latex = function(){ return '{{var:' + this.ends[L].latex() + '}}'; };
-  _.text = function(){ return this.ends[L].text(); };
+  _.latex = function() {
+    return this.text();
+  };
+  _.text = function() {
+    return '{{var:' + this.jQ.find('[mathquill-command-id]').text() + '}}';
```

Comment:
I've taken this formatting logic from [`renderMathDynamicVars()`](https://github.com/Learnosity/api-questioneditor/blob/fb2047700fd3c92c40a924fc1f86b77dc7a15b25/www/latest/app/views/shared/questionDynamicContentMixin.js#L50).

Reply:
Hey @scott-hughes, in general it's an antipattern to rely on the DOM rather than the internal mq math tree for this sort of stuff - in my experience it usually means you're doing something wrong.

Reply:
Good catch – yes, I completely agree with that.

For reference, I tried two other alternatives:

1. Call `this.ends[L].latex()` but remove the math formatting commands with a regex replacement (as shown in 4c14a70e0c7988f2478b0dd064f541bb56f7db98).
2. Inherit from `TextBlock` rather than `MathCommand`. I spent about 30 minutes on this but couldn't get it working due to my lack of familiarity with the MathQuill codebase.

Ultimately, I think your suggestion of tackling this bug at a different level is the best approach for now, so that's what I've gone with.

Reply:
Nice one, thanks @scott-hughes. Yeah there's a bit of a learning curve on this one alright!